http://dev.mutt.org/hg/mutt/rev/817c60d5da7d

# HG changeset patch
# User Brendan Cully <brendan@kublai.com>
# Date 1268089116 28800
# Node ID 817c60d5da7dae3376a6ad683091ae24d7cf2ed2
# Parent  8ec606f6cafb55299bec617d7d743deceb8d9e2d
mutt_buffer_printf: NULL dptr means use data

diff -r 8ec606f6cafb -r 817c60d5da7d muttlib.c
--- a/muttlib.c	Mon Mar 08 14:52:17 2010 -0800
+++ b/muttlib.c	Mon Mar 08 14:58:36 2010 -0800
@@ -1675,6 +1675,9 @@
   va_start (ap, fmt);
   va_copy (ap_retry, ap);
 
+  if (!buf->dptr)
+    buf->dptr = buf->data;
+
   doff = buf->dptr - buf->data;
   blen = buf->dsize - doff;
   /* solaris 9 vsnprintf barfs when blen is 0 */

